src/app/convert/css-scss/page.jsx

```diff
@@ -0,0 +1,313 @@
+"use client";
+
+import React, { useState, useEffect } from "react";
+import * as sass from "sass"
+import SnippetBox from "./components/SnippetBox";
+import ResultBox from "./components/ResultBox";
+import Nav from "@/app/components/nav";
+import Footer from "@/app/components/Footer";
+import Image from "next/image";
+import icon from "./switch_icon.png";
+import { forEach } from "jszip";
+
+export default function CSS_TO_SCSS() {
+  //only nav content
+  const [isDarkMode, setIsDarkMode] = useState(false);
+
+  useEffect(() => {
+    if (isDarkMode) {
+      document.documentElement.classList.add("dark");
+    } else {
+      document.documentElement.classList.remove("dark");
+    }
+  }, [isDarkMode]);
+
+  const toggleTheme = () => {
+    setIsDarkMode(!isDarkMode);
+  };
+
+  //page content
+
+  const [snippet, setSnippet] = useState("");
+  const [resultSnippet, setResultSnippet] = useState("");
+  const [mode, setMode] = useState({ from: "css", to: "scss" });
+
+  //CSStoSCSS converter function
+  const cssToScss = (css) => {
```

Comment:
It is preferable to use a library. Because i have tried this and it generated it incorrectly.

Reply:
Let me know which library you want to use to convert css to scss because i didn't find any

Reply:
You can use the official sass library. Link: https://sass-lang.com/documentation/js-api/

Reply:
Sass library doesn't have a css to scss file or snippet converter. It only works from sass or scss to css.

This is the closest library of css to scss: https://github.com/feihe08/css2sass
It does the same I am doing in the code, regex logic and output accordingly. The difference is that it only returns sass and the conversion only works on files not snippets.

If you try it and didn't work, feel free to give me the snippet you test on the converter.

Reply:
I think for now you can just do it as css to sass.